core/src/com/unciv/UncivGame.kt

```diff
@@ -274,6 +278,14 @@ class UncivGame(parameters: UncivGameParameters) : Game() {
     companion object {
         lateinit var Current: UncivGame
         fun isCurrentInitialized() = this::Current.isInitialized
+        fun isCurrentGame(gameId: String): Boolean {
+            return if (isCurrentInitialized() && Current.isGameInfoInitialized()) {
+                Current.gameInfo.gameId == gameId
+            } else {
+                false
+            }
+
+        }
```

Comment:
Sounds to me like 'return isCurrentInitialized() && Current.isGameInfoInitialized() && Current.gameInfo.gameId == gameId`

Reply:
Whenever you have an if/else where one of the options is just a boolean you know it can be simplified

Reply:
> need to get rid of the `lateinit` modifier

Oh, yes, wherever possible or impossible. And the God Object:tm: itself too! (Actually, it's not so bad)